openmessaging/src/main/java/io/openmessaging/rocketmq/consumer/PullConsumerImpl.java

```diff
@@ -76,110 +93,268 @@ public PullConsumerImpl(final KeyValue properties) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public List<String> getBindQueues() {
+        Set<String> registerTopics = rocketmqPullConsumer.getRegisterTopics();
+        return new ArrayList<>(registerTopics);
+    }
+
+    @Override
+    public void addInterceptor(ConsumerInterceptor interceptor) {
+        consumerInterceptors.add(interceptor);
+    }
+
+    @Override
+    public void removeInterceptor(ConsumerInterceptor interceptor) {
+        consumerInterceptors.remove(interceptor);
+    }
+
+    @Override
+    public Message receive(long timeout) {
+        KeyValue properties = new DefaultKeyValue();
+        properties.put("TIMEOUT", timeout);
+        MessageExt rmqMsg = localMessageCache.poll(properties);
         return rmqMsg == null ? null : OMSUtil.msgConvert(rmqMsg);
     }
 
     @Override
-    public Message receive(final KeyValue properties) {
+    public Message receive(String queueName, int partitionId, long receiptId, long timeout) {
+        KeyValue properties = new DefaultKeyValue();
+        properties.put("QUEUE_NAME", queueName);
+        properties.put("PARTITION_ID", partitionId);
+        properties.put("RECEIPT_ID", receiptId);
+        properties.put("TIMEOUT", timeout);
         MessageExt rmqMsg = localMessageCache.poll(properties);
         return rmqMsg == null ? null : OMSUtil.msgConvert(rmqMsg);
     }
 
     @Override
-    public void ack(final String messageId) {
-        localMessageCache.ack(messageId);
+    public List<Message> batchReceive(long timeout) {
+        KeyValue properties = new DefaultKeyValue();
+        properties.put("TIMEOUT", timeout);
+        List<MessageExt> rmqMsgs = localMessageCache.batchPoll(properties);
+        if (null != rmqMsgs && !rmqMsgs.isEmpty()) {
+            List<Message> messages = new ArrayList<>(rmqMsgs.size());
+            for (MessageExt messageExt : rmqMsgs) {
+                BytesMessageImpl bytesMessage = OMSUtil.msgConvert(messageExt);
+                messages.add(bytesMessage);
+            }
+            return messages;
+        }
+        return null;
     }
 
     @Override
-    public void ack(final String messageId, final KeyValue properties) {
-        localMessageCache.ack(messageId);
+    public List<Message> batchReceive(String queueName, int partitionId, long receiptId, long timeout) {
+        MessageQueue mq = null;
+        try {
+            Set<MessageQueue> messageQueues = rocketmqPullConsumer.fetchSubscribeMessageQueues(queueName);
+            for (MessageQueue messageQueue : messageQueues) {
+                if (messageQueue.getQueueId() == partitionId) {
+                    mq = messageQueue;
+                }
+            }
+        } catch (MQClientException e) {
+            e.printStackTrace();
+        }
+        PullResult pullResult;
+        try {
+            pullResult = rocketmqPullConsumer.pull(mq, "*", receiptId, 4 * 1024 * 1024, timeout);
```

Comment:
Here,"4 * 1024 * 1024" you can add this contents into constants class.

Reply:
It has been modified